keps/sig-node/20190517-image-decryption.md

```diff
@@ -0,0 +1,314 @@
+---
+title: Adding the Support for Encrypted Images
+authors:
+  - "@harche"
+owning-sig: sig-node
+participating-sigs:
+  - sig-architecture
+reviewers:
+  - smarterclayton
+  - tallclair
+  - yujuhong
+approvers:
+  - smarterclayton
+  - tallclair
+creation-date: 2019-05-16
+status: provisional
+---
+
+# Adding the Support for Encrypted Images
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non\-Goals](#non-goals)
+  * [User Stories](#user-stories)
+* [Proposal](#proposal)
+  * [API](#api)
+    * [Image Handler](#Image-handler)
+  * [Consumption of the ImageDecryptSecrets](#Consumption-of-the-ImageDecryptSecrets)
+* [Alternatives Considered](#alternatives-considered)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+The underlying specification for the containers, the OCI spec, is soon going to support encrypted images. Kubernetes should be able to support decryption of these encrypted images with the addition of a new type of `Secret`, which we would like to call `ImageDecryptSecret`. 
```

Comment:
It seems this is still up for debate? skimming the upstream issue / PR it seems like there hasn't been much recent discussion and I don't see agreement on this issue yet?

Reply:
Hi @BenTheElder! There has been ongoing discussions around the OCI spec outside the issue for the past year. We had bi-weekly online working sessions  for the most part from the start of the year. The discussion team around this include @stevvooe (OCI maintainer) as well as @estesp and @dmcgowan (containerd maintainers), the team from IBM (@stefanberger, @lumjjb, @harche) and @justincormack from Docker. 

Currently where we are is that we have agreement on the general construct, but we are figuring out the finer details. The awesome thing about the KEP is that it is actually independent on the details of the OCI/runtime, as `ImagePullSecrets` are to the OCI image spec. :)